common.go

```diff
@@ -0,0 +1,37 @@
+package set
+
+// CommonSet is the interface that all sets implement
+type CommonSet[T any] interface {
```

Comment:
@zonewave I was thinking about renaming this to `Interface` - similar to the style of Go's [sort.Interface](https://pkg.go.dev/sort#Interface). That way it avoids using the word "set" twice in external packages (e.g. `set.Interface` instead of `set.CommonSet`) (or maybe `set.Common` would be nice?)